### PR TITLE
update quick start notebook

### PR DESCRIPTION
**Pull request by yinxi-db**
### Related Issues/PRs
none

### What changes are proposed in this pull request?

use the new 5 min agent demo notebook with synthetic data generation API for the quick start demo

### How is this PR tested?
successful run [here](https://e2-demo-field-eng.cloud.databricks.com/editor/notebooks/3280716146143522?o=1444828305810485)

### CODE REVIEW

File: quick_start_demo/5-min-Mosaic-AI-Agents-Demo-with-Synthetic-Eval.py [hunk 1]
```diff
@@ -0,0 +1,699 @@
...
+# MAGIC %autoreload 2
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## 0.1/ Configure user name to avoid conflicts with other users running the demo
+
+# COMMAND ----------
+
+import mlflow
```
Comment (epec254) [inline quick_start_demo/5-min-Mosaic-AI-Agents-Demo-with-Synthetic-Eval.py:51]: Does this add value if the user has to configure the catalog/schema anyways?

I'd opt towards removing.

File: quick_start_demo/5-min-Mosaic-AI-Agents-Demo-with-Synthetic-Eval.py [hunk 2]
```diff
@@ -0,0 +1,699 @@
...
+
+print(f"User name: {user_name}")
+
+# COMMAND ----------
+
+########################
+# #### ✅✏️ CONFIGURE THE UC MODEL NAME HERE
+########################
+#UC_MODEL_NAME = f"<ENTER_YOUR_CATALOG_NAME>.<ENTER_YOUR_SCHEMA_NAME>.db_docs__{user_name}"
+UC_MODEL_NAME = f"yz_agent_demo.playground.db_docs__{user_name}"
```
Comment (epec254) [inline quick_start_demo/5-min-Mosaic-AI-Agents-Demo-with-Synthetic-Eval.py:67]: rename this to 'catalog.schema.my_demo_agent' to improve the clarity

File: quick_start_demo/5-min-Mosaic-AI-Agents-Demo-with-Synthetic-Eval.py [hunk 3]
```diff
@@ -0,0 +1,699 @@
...
+# MAGIC * Permission to create Model Serving endpoints
+# MAGIC
+# MAGIC **How to use this notebook**:
+# MAGIC 1. Update the following cell to configure the UC location where the agent model will be register to
+# MAGIC
+# MAGIC  ✅✏️ CONFIGURE THE UC MODEL NAME HERE
+# MAGIC
+# MAGIC 2. Click Run all
+# MAGIC
+# MAGIC ![alt text](https://github.com/databricks/genai-cookbook/blob/demo_gifs/quick_start_demo/demo_gifs/demo_overview.gif?raw=true "Title")
```
Comment (epec254) [inline quick_start_demo/5-min-Mosaic-AI-Agents-Demo-with-Synthetic-Eval.py:25]: @peter lu has new gifs for all the gifs throughout the notebook - can we use those?

File: quick_start_demo/5-min-Mosaic-AI-Agents-Demo-with-Synthetic-Eval.py [hunk 4]
```diff
@@ -0,0 +1,699 @@
...
+# MAGIC from openai import OpenAI
+# MAGIC import pandas as pd
+# MAGIC from typing import Any, Union, Dict, List
+# MAGIC import mlflow
+# MAGIC from mlflow.models.rag_signatures import ChatCompletionResponse, ChatCompletionRequest
+# MAGIC import dataclasses
+# MAGIC import json
+# MAGIC
+# MAGIC DEFAULT_CONFIG = {
+# MAGIC     'endpoint_name': "agents-demo-gpt4o",
```
Comment (epec254) [inline quick_start_demo/5-min-Mosaic-AI-Agents-Demo-with-Synthetic-Eval.py:193]: let's change this to FMAPI llama 70b

File: quick_start_demo/5-min-Mosaic-AI-Agents-Demo-with-Synthetic-Eval.py (hunk 2 above)
Comment (yinxi-db) [reply quick_start_demo/5-min-Mosaic-AI-Agents-Demo-with-Synthetic-Eval.py]: agree, updated

File: quick_start_demo/5-min-Mosaic-AI-Agents-Demo-with-Synthetic-Eval.py (hunk 1 above)
Comment (yinxi-db) [reply quick_start_demo/5-min-Mosaic-AI-Agents-Demo-with-Synthetic-Eval.py:51]: agree, removed the user name parsing, but keep the workspace client

File: quick_start_demo/5-min-Mosaic-AI-Agents-Demo-with-Synthetic-Eval.py (hunk 4 above)
Comment (yinxi-db) [reply quick_start_demo/5-min-Mosaic-AI-Agents-Demo-with-Synthetic-Eval.py]: we need to re-think about the LLM endpoint choices in this notebook.
I have changed the default config in `fc_agent.py` and also the `dev_config` of the vibe check section. However, the baseline config still has `"endpoint_name": "agents-demo-gpt4o""agents-demo-gpt4o"`. 
More importantly, it makes me realize that the notebook requires the user have 2 external model endpoints (`agents-demo-gpt4o` and `agents-demo-gpt4o-mini`) to succeed the "Run All". That should be mentioned in the requirements and 0/setup. I think it is better to remove the external model dependencies, however, all the screenshots show gpt models.

File: quick_start_demo/5-min-Mosaic-AI-Agents-Demo-with-Synthetic-Eval.py
Comment (yinxi-db) [inline quick_start_demo/5-min-Mosaic-AI-Agents-Demo-with-Synthetic-Eval.py:1]: requires users to have a deployed gpt-4o endpoint with name `agents-demo-gpt4o`. do we want to use workspace foundation models like llama 3.1 405b or DBRX

Comment (yinxi-db) [inline quick_start_demo/5-min-Mosaic-AI-Agents-Demo-with-Synthetic-Eval.py:1]: need to update run name if endpoint changed

Comment (yinxi-db) [inline quick_start_demo/5-min-Mosaic-AI-Agents-Demo-with-Synthetic-Eval.py:1]: gif needs to be updated if endpoint is changed

File: quick_start_demo/5-min-Mosaic-AI-Agents-Demo-with-Synthetic-Eval.py [hunk 5]
```diff
@@ -0,0 +1,694 @@
...
+# MAGIC - See the judge outputs to understand why the responses were graded as good/bad
+# MAGIC - Compare between multiple runs to see how quality changed between experiments
+# MAGIC
+# MAGIC You can also inspect the other tabs:
+# MAGIC - `Overview` lets you see the Agent's config/parameters
+# MAGIC - `Artifacts` lets you see the Agent's code
+# MAGIC
+# MAGIC This UIs, coupled with the speed of evaluation, help you efficiently test your hypotheses to improve quality, letting you reach the production quality bar in less time. 
+# MAGIC
+# MAGIC ![alt text](https://github.com/databricks/genai-cookbook/blob/demo_gifs/quick_start_demo/demo_gifs/eval_1.gif?raw=true "Title")
```
Comment (yinxi-db) [inline quick_start_demo/5-min-Mosaic-AI-Agents-Demo-with-Synthetic-Eval.py:510]: need to update the gif if dev config uses llama-70b

File: quick_start_demo/5-min-Mosaic-AI-Agents-Demo-with-Synthetic-Eval.py [hunk 6]
```diff
@@ -0,0 +1,694 @@
...
+            model=model_info.model_uri,  # Logged Agent from above
+            model_type="databricks-agent",  # activate Mosaic AI Agent Evaluation
+        )
+
+        return (model_info, eval_results)
+
+
+# Now we call the helper function to run evaluation.
+# The configuration keys must match those defined in `fc_agent.py`
+model_info_gpt4o, eval_results = log_and_evaluate_agent(
```
Comment (yinxi-db) [inline quick_start_demo/5-min-Mosaic-AI-Agents-Demo-with-Synthetic-Eval.py:552]: requires a deployed gpt-4o endpoint with name agents-demo-gpt4o in the workspace. do we want to use workspace foundation models like llama 3.1 405b or DBRX
will need to update variable name, endpoint name, and run name if the endpoint config changes

File: quick_start_demo/5-min-Mosaic-AI-Agents-Demo-with-Synthetic-Eval.py [hunk 7]
```diff
@@ -0,0 +1,694 @@
...
+# MAGIC
+# MAGIC Then, go to the MLflow Experiement page and click the chart icon in the upper left corner by `Runs`.  Here, you can compare the models quantiatively across quality/cost/latency metrics.  Cost is proxied through the number of tokens used.
+# MAGIC
+# MAGIC This helps you make informed tradeoffs in partnership with your business stakeholders about cost/latency/quality.  Further, you can use this view to provide quantitative updates to your stakeholders so they can follow your progress improving quality!
+# MAGIC
+# MAGIC ![alt text](https://github.com/databricks/genai-cookbook/blob/demo_gifs/quick_start_demo/demo_gifs/eval_2.gif?raw=true "Title")
+
+# COMMAND ----------
+
+baseline_config = {
```
Comment (yinxi-db) [inline quick_start_demo/5-min-Mosaic-AI-Agents-Demo-with-Synthetic-Eval.py:591]: same comments here, need to decide whether or not to remove external models in the notebook

**Review by epec254 — APPROVED** [review]

**Pull request merged**